### PR TITLE
ANCM: Detect client disconnects and logging

### LINKED ISSUE #9141: ANCM: Detect client disconnects and logging
https://blogs.msdn.microsoft.com/webdev/2018/10/17/asp-net-core-2-2-0-preview3-now-available/#aspnetcoremodule

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: fe69b3a9-896f-e189-0c2c-f87ff735df9c
* Version Independent ID: 97c8549b-66ad-0a23-1971-784fb923042a
* Content: [ASP.NET Core Module configuration reference](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/aspnet-core-module?view=aspnetcore-2.1)
* Content Source: [aspnetcore/host-and-deploy/aspnet-core-module.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/host-and-deploy/aspnet-core-module.md)
* Product: **aspnet-core**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #9141 

Carrying over the question on ...

> Debug level (`debugLevel`) values:
>
> \* ERROR
> \* WARNING
> \* INFO
> \* TRACE
> \* CONSOLE
> \* FILE
> \* EVENTLOG

The first four look like logging levels, and the last three look like locations. How do I fix the coverage?

### CODE REVIEW

File: aspnetcore/host-and-deploy/aspnet-core-module.md [hunk 1]
```diff
@@ -303,6 +305,47 @@ The following sample `aspNetCore` element configures stdout logging for an app h
...
+* INFO
+* TRACE
+* CONSOLE
+* FILE
+* EVENTLOG
+
+The handler settings can also be provided via environment variables:
+
+* `ASPNETCORE_MODULE_DEBUG_FILE` &ndash; Path to the debug log file.
+  * IIS default: *aspnetcore-debug.log*
```
Comment (jkotalik) [inline aspnetcore/host-and-deploy/aspnet-core-module.md]: Shouldn't the file names be the same here?

Comment (guardrex) [reply aspnetcore/host-and-deploy/aspnet-core-module.md]: I thought I saw that it defaults differently. I'll go back into the repo and see where I saw these and let you know.

Comment (guardrex) [reply aspnetcore/host-and-deploy/aspnet-core-module.md]: Found it ... it came from here ...

https://github.com/aspnet/AzureIntegration/blob/master/extensions/Microsoft.AspNetCore.Runtime.SiteExtension/applicationHost.xdt

**Comment by guardrex** [comment]
What about what I ask in the OP :point_up: ... what do I say about "levels" and "locations" being valid for the `debugLevel` value?

File: aspnetcore/host-and-deploy/aspnet-core-module.md (hunk 1 above)
Comment (guardrex) [reply aspnetcore/host-and-deploy/aspnet-core-module.md]: Is that only for the site extension? ... looks that way.

**Comment by jkotalik** [comment]
I'd just say that the debugLevel is used to specify both the level and destination for logs. You can specify more than one value for the log location too. 

Log Levels:
* ERROR
* WARNING
* INFO
* TRACE

Log Location:
* CONSOLE
* FILE
* EVENTLOG

File: aspnetcore/host-and-deploy/aspnet-core-module.md (hunk 1 above)
Comment (jkotalik) [reply aspnetcore/host-and-deploy/aspnet-core-module.md]: This is only for the site extension. This isn't inserted by default for azure deployments. @pakrym to confirm.

Comment (jkotalik) [reply aspnetcore/host-and-deploy/aspnet-core-module.md]: Aka it will still be *aspnetcore-debug.log* in azure.

Comment (guardrex) [reply aspnetcore/host-and-deploy/aspnet-core-module.md]: I'll stand-by on this one until we hear back. If it's set for the site extension as ...

```
%HOME%\LogFiles\Application\amcm.log
```
... and stdout logs to go to ...

```
\\?\%home%\LogFiles\
```
... there might be some lack of consistency in providing no path (the root of the deployment) by just showing `aspnetcore-debug.log`.

**Comment by jkotalik** [comment]
See https://github.com/aspnet/AzureIntegration/issues/242. Will address that shortly.

**Comment by jkotalik** [comment]
See: https://github.com/aspnet/AzureIntegration/issues/242#issuecomment-432457703

**Comment by jkotalik** [comment]
🚢 🇮🇹

**Pull request merged**